### DISCUSSION TITLE
Servo Trigger on a define Waypoint Number in Programming

### DISCUSSION THREAD

**Discussion by synersignart** [iNavFlight/inav#8684] [General]
<html><body>
in Programming is there anyway we indicate the Waypoint number like WP_1 , WP_2, WP_3.....
one such use is to trigger a servo when the drone is at a waypoint number.

can someone Explain the difference and its Values of
Current_Waypoint_Index and Current_Waypoint_Action
I cant find any clear documenation stating what it actually does 
typedef enum {
    NAV_WP_ACTION_WAYPOINT  = 0x01,
    NAV_WP_ACTION_HOLD_TIME = 0x03,
    NAV_WP_ACTION_RTH       = 0x04,
    NAV_WP_ACTION_SET_POI   = 0x05,
    NAV_WP_ACTION_JUMP      = 0x06,
    NAV_WP_ACTION_SET_HEAD  = 0x07,
    NAV_WP_ACTION_LAND      = 0x08
} navWaypointActions_e;

also whats odd is the missing number 2 on the Active Waypoint Action

<h4 dir="auto"><a id="user-content-active_waypoint_action" class="anchor" aria-hidden="true" href="https://github.com/iNavFlight/inav/blob/master/docs/Programming%20Framework.md#active_waypoint_action"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"></svg>

<html><body>

ACTIVE_WAYPOINT_INDEX | Indexed from 1. To verify WP is in progress, use IS_WP
-- | --

</a>ACTIVE_WAYPOINT_ACTION</h4>

</body>
</html>

Action | Value
-- | --
WAYPOINT | 1
HOLD_TIME | 3
RTH | 4
SET_POI | 5
JUMP | 6
SET_HEAD | 7
LAND | 8

</body>
</html>

**Comment by MrD-RC** [comment]
The question about action 2 was answered in your other discussion. 

There will hopefully be more options in INAV 6.0 for triggering things based on the waypoint or, better for repeated triggers, user actions.
